.github/workflows/ci.yml

```
@@ -210,6 +212,11 @@ jobs:
           # Note: it appears fail below min is one off, use fail_under -1 here
           thresholds: '95 98'
 
+      - name: Add link to report badge
```

To not skip the step if coverage report raise on fail under. Possibly condition on markdown existence instead 

```suggestion
      - name: Add link to report badge
        if: ${{ always() && steps.cov_xml_upload.outputs.artifact-id != '' }}
```